test/test_mobile_optimizer.py

```diff
@@ -207,5 +208,88 @@ def get_lint_count_by_type(lint_type, module_lint_List):
         bi_module_lint_list = generate_mobile_module_lints(bi_module)
         self.assertEqual(len(bi_module_lint_list), 0)
 
+    @unittest.skipUnless(torch.backends.xnnpack.enabled,
+                         " XNNPACK must be enabled for these tests."
+                         " Please build with USE_XNNPACK=1.")
+    def test_hoist_conv_packed_params(self):
+
+        class Standalone(nn.Module):
+            def __init__(self):
+                super(Standalone, self).__init__()
+                self.quant = torch.quantization.QuantStub()
+                self.conv1 = nn.Conv2d(1, 1, 1)
+                self.dequant = torch.quantization.DeQuantStub()
+
+            def forward(self, x):
+                x = self.quant(x)
+                x = self.conv1(x)
+                x = self.dequant(x)
+                return x
+
+        class Child(nn.Module):
+            def __init__(self):
+                super(Child, self).__init__()
+                self.conv1 = nn.Conv2d(1, 1, 1)
+
+            def forward(self, x):
+                x = self.conv1(x)
+                return x
+
+        class Parent(nn.Module):
+            def __init__(self):
+                super(Parent, self).__init__()
+                self.quant = torch.quantization.QuantStub()
+                self.conv1 = nn.Conv2d(1, 1, 1)
+                self.child = Child()
+                # TODO: test nn.Sequential after #42039 is fixed
+                self.dequant = torch.quantization.DeQuantStub()
+
+            def forward(self, x):
+                x = self.quant(x)
+                x = self.conv1(x)
+                x = self.child(x)
+                x = self.dequant(x)
+                return x
+
+        def _static_quant(model):
+            model.qconfig = torch.quantization.get_default_qconfig('qnnpack')
+            torch.quantization.prepare(model, inplace=True)
+            model(torch.randn(4, 1, 4, 4))
+            torch.quantization.convert(model, inplace=True)
+            return model
+
+        # basic case
+
+        m = Standalone()
+        m = _static_quant(m)
+        m = torch.jit.script(m)
+        m_optim = optimize_for_mobile(m)
+
+        FileCheck().check_not("Conv2d = prim::GetAttr[name=\"conv1\"]") \
```

Comment:
also can we check the access to conv module is removed?

Reply:
sure, will add

Reply:
actually, sorry, what do you mean specifically?  The original access is 
```
__torch__.torch.nn.quantized.modules.conv.___torch_mangle_13.Conv2d = prim::GetAttr[name="conv1"](%self)
```

I'm checking for the absence of a suffix of that (for readability).  Happy to adjust if something else is better

Reply:
I mean check the conv attribute is removed from the parent module

Reply:
makes sense.  Any tips on the best way to do that with `FileCheck`?

Reply:
FileCheck is just checking the strings. for this one we need to check the module itself with something like
`self.assertFalse(hasattr(module, 'conv1'))`